src/middleware.ts

```diff
@@ -122,6 +122,27 @@ export async function middleware(req: NextRequest, ev: NextFetchEvent) {
 	// 	}
 	// }
 
+	/**
+	 * Detect the variants query param and rewrite to the correct path.
+	 *
+	 * Request path: /{product}/tutorials/{collection}/{tutorial}?variants={slug:optionSlug}
```

Comment:
I chose to go plural here in anticipation of multi-variant world. Open to make this `variant` though if folks have a strong opinion